compiler/noirc_printable_type/src/lib.rs

```diff
@@ -282,11 +290,17 @@ impl<F: AcirField> std::fmt::Display for PrintableValueDisplay<F> {
                 let mut display_iter = values.iter();
                 let re = Regex::new(r"\{([a-zA-Z0-9_]+)\}").map_err(|_| std::fmt::Error)?;
```

Comment:
This could also be done by using a regex that matches `{...}` but not `{{...}}` but I couldn't figure out how to do it so instead I check if there's `{` before each match.

Reply:
Hmm... I'm realizing that, like this, this program doesn't work as expected:

```rust
fn main() {
    let hello_world = 1234;
    let x = f"{{{hello_world}}}";
    println(x);
}
```

It prints "{{hello_world}}" instead of interpolating it, because there's "{" before "{hello_world}". I'm not sure how to solve this... maybe if there's another "{" before it we do need to interpolate it (but not if there's yet one more "{" before it). I'll try to fix it.

Reply:
It worked!